app/sqaws.py

```diff
@@ -306,50 +333,23 @@ def terminate_resource(self, region_name: str, resource_id: str, dryrun: bool) -
             print(f'Failure when calling terminate_instances: {str(e)}')
             return False
 
-    # Create a list of all stoppable instances
-    def get_stoppable_resources(self, instances):
-        return list(i for i in instances if self.is_stoppable(i))
-
-    # Create a list of all terminatable instances
-    def get_terminatable_resources(self, resources):
-        return list(i for i in resources if self.is_terminatable(i))
-
-    # Check if an instance is stoppable
-    @staticmethod
-    def is_stoppable(instance, is_weekend=TODAY_IS_WEEKEND):
-        parsed_date: parsing.ParsedDate = parsing.parse_date_tag(instance.stop_after)
-
-        return instance.state == 'running' and (
-            # Treat unspecified "Stop after" dates as being in the past
-            (parsed_date.expiry_date is None and not parsed_date.on_weekends)
-            or (parsed_date.on_weekends and is_weekend)
-            or (parsed_date.expiry_date is not None and TODAY_YYYY_MM_DD >= parsed_date.expiry_date))
-
     # Check if an instance is terminatable
-    def is_terminatable(self, resource):
-        parsed_date: parsing.ParsedDate = parsing.parse_date_tag(resource.terminate_after)
-
-        # For now, we'll only terminate instances which have an explicit 'Terminate after' tag
-        return resource.state == 'stopped' and (
-            (parsed_date.expiry_date is not None and TODAY_YYYY_MM_DD >= parsed_date.expiry_date))
-
-    # Check if an instance is safe to stop
-    def is_safe_to_stop(self, instance, is_weekend=TODAY_IS_WEEKEND):
-        warning_date = parsing.parse_date_tag(instance.stop_after).warning_date
-        return self.is_stoppable(instance, is_weekend=is_weekend) \
-            and warning_date is not None and warning_date <= TODAY_YYYY_MM_DD
+    @staticmethod
+    def is_terminatable(resource, today_date):
+        state = 'running'
+        return Resource.generic_is_terminatable(resource, state, today_date)
```

Comment:
Why use a `@staticmethod` for this case? Wouldn't

```
    def is_terminatable(self, today_date):
        state = 'running'
        return Resource.generic_is_terminatable(self, state, today_date)
```

accomplish the same thing in a more idiomatic class way?

Reply:
Just pushed another commit that changes this (and several other) methods to use self and not be static - you're right, it makes more sense that way